dvc/path_info.py

```diff
@@ -241,3 +241,79 @@ class CloudURLInfo(URLInfo):
     @property
     def path(self):
         return self._spath.lstrip("/")
+
+
+class HTTPURLInfo(URLInfo):
+    def __init__(self, url):
+        p = urlparse(url)
+        stripped = p._replace(params=None, query=None, fragment=None)
+        super().__init__(stripped.geturl())
```

Comment:
More future-proof to use `super()` to use the parent's logic. Otherwise we'd need

```python
p = urlparse(url)
assert p.password is None
self.fill_parts(p.scheme, p.hostname, p.username, p.port, p.path)
```

and not call `super()`, which is allowed but not great practice.

Reply:
First, we can ignore all of this as long as it works. Some tech debt, but might be resolved later.

Needing to parse, restringify and parse is a kludge.